### ISSUE TITLE
Authentication failed

### ISSUE THREAD

**Issue by dppeak** [black-ops-drivers/control4-myq#11]
An error occurred refreshing devices statuses; missing email and/or password

**Comment by brianbmwboy** [comment]
Same Here. HC800 3.3.0  .  Do we need to be on 3.3.1 or only ea, core controller?

**Comment by dppeak** [comment]
I am on 3.4

**Comment by dppeak** [comment]
I think it's myQ people changing things again so they can charge for subscriptions.

**Comment by dppeak** [comment]
Yeah, I had been running with the Homebridge-myq plugin to access through HomeKit.  It started failing badly this week.  Was hoping this was working without issues. :(

**Comment by black-ops-drivers** [comment]
It’s possible this plugin will still work, but I need to reproduce the issue. The bot protection is ML based so the more users the more easily it can block it. I will take a look and see if this is just a simpler fix since this integration doesn’t have a ton of users.

**Comment by brianbmwboy** [comment]
Is this compatible with hc800 and 3.3.0? And should we expect interruptions from Myq on their end blocking this frequently?

**Comment by jaas666** [comment]
> It’s possible this plugin will still work, but I need to reproduce the issue. The bot protection is ML based so the more users the more easily it can block it. I will take a look and see if this is just a simpler fix since this integration doesn’t have a ton of users.

I believe cloudflare is presenting the recaptcha for the requests thus our authentication calls fail.

Maybe finding a way of presenting the recaptcha page to the end user could work?

**Comment by black-ops-drivers** [comment]
Yeah I have tried that, but the resulting code used to exchange for a token results in a 401 response.

**Comment by jaas666** [comment]
Ahh that blows.
Sincerely I don't know what they win in doing this. I think more people
would buy their products if they were compatible with home automation
platforms.. either by or using API access or doing their own integrations.
Makes no sense to me.

**Comment by black-ops-drivers** [comment]
Yeah pretty lame. I’ll continue looking into a solution when I have time.
